src/compiler/crystal/compiler.cr

```diff
@@ -405,6 +424,19 @@ module Crystal
       end
     end
 
+    def optimization_mode_suffix
+      case @optimization_mode
+      when OptimizationMode::O3
+        ".o3"
+      when OptimizationMode::O2
+        ".o2"
+      when OptimizationMode::O1
+        ".o1"
+      else
+        ".o0"
+      end
+    end
```

Comment:
**suggestion:** Maybe this method would fit better as `OptimizationMode#suffix`. That feels more idiomatic.
Implementation could also use exhaustive case and predicate methods.
Or even more concise, just use stringification: `".#{to_s.downcase}"`.